doc/pages/Manual.md

```diff
@@ -7,33 +7,228 @@ primarily of use for packagers, package maintainers and repository maintainers.
-is progressively being updated with the new features of opam `2.0`.
+```
+opam repository add <name> <URL>
+```
+
+and can subsequently be then selected for use in specific switches using `opam
+repository select <name>`. Use `opam repository list --all` for an overview of
+configured repositories. Repository selection is always ordered, with the
+definition of a given version of a package being taken from the repository with
+the lowest index where it is found.
+
+Data from the configured repositories is updated from the upstreams manually
+using the `opam update` command.
+
+### Switches
+
+Opam is designed to hold any number of concurrent installation prefixes, called
+_switches_. Switches are isolated from each other and have their own set of
+installed packages, selection of repositories, and configuration options. All
+package-related commands operate on a single switch, and require one to be
+selected.
+
+The current switch can be selected in the following ways:
+- globally, using `opam switch <switch>`. Opam will use that switch for all
+  further commands, except when overriden in one of the following ways.
+- for local switches, which are external to the opam root, when in the directory
+  where the switch resides or a descendant.
+- through the `--switch <switch>` command-line flag, for a single command.
+- by setting the `OPAMSWITCH=<switch>` environment variable, to set it within a
+  single shell session. This can be done by running `eval $(opam config env
+  --switch <switch>)` to set the shell environment at once, see below.
+
+Switches have their own prefix, normally `~/.opam/<switch>`, where packages get
+intalled ; to use what is installed in a switch, some environment variables need
+to be set, _e.g._ to make executables installed into `~/.opam/<switch>/bin`
+visible, that directory needs to be added to `PATH`, but individual packages can
+define their own settings as well.
+
+Command `opam config env` returns the environment updates corresponding to the
+current switch, in a format readable by your shell, and when needed opam will
+prompt you to run:
+
+```
+eval $(opam config env)
+```
 
-## General file format
+A switch is created using `opam switch create <switch> (<compiler>|--empty)`.
+- `<switch>` can be either a plain name, or a directory name (if containing `/`
+  or starting with `.`). In the latter case the switch is _local_ and instead of
+  being held at `~/.opam/<switch>`, it will be created in the given directory,
+  as a `_opam` subdirectory. Local switches are automatically selected depending
+  on the current directory, see above.
+- If a `<compiler>` is selected, opam will install the corresponding packages
+  and their dependencies in the new switch. These packages will be marked as
+  _base_, protected against removal and unaffected by upgrade commands.
+  `<compiler>` can be selected among packages which have the `compiler` flag
```

Comment:
indeed, `base` and `compiler` are almost synonyms and get a bit mixed up, the story could be clearer.
Dependencies of your `compiler` package get installed at switch init, and are marked as `base` too.